integrations/watsonx/src/haystack_integrations/components/generators/watsonx/chat/chat_generator.py

```diff
@@ -0,0 +1,391 @@
+        :param verify: SSL verification setting. Can be:
+            - True: Verify SSL certificates (default)
+            - False: Skip verification (insecure)
+            - Path to CA bundle for custom certificates
+        """
+        self.api_key = api_key or Secret.from_env_var("WATSONX_API_KEY")
+        self.model = model
+        self.project_id = project_id or Secret.from_env_var("WATSONX_PROJECT_ID")
+        self.api_base_url = api_base_url or "https://us-south.ml.cloud.ibm.com"
+        self.generation_kwargs = generation_kwargs or {}
+        self.timeout = timeout or float(os.environ.get("WATSONX_TIMEOUT", "30.0"))
+        self.max_retries = max_retries or int(os.environ.get("WATSONX_MAX_RETRIES", "5"))
+        self.tools = tools
+        self.verify = verify
+
+        if not project_id:
+            msg = "project_id must be provided"
+            raise ValueError(msg)
+
+        self._initialize_client()
+
+    def _initialize_client(self):
+        """Initialize the Watsonx client with configured credentials."""
+        credentials = Credentials(api_key=self.api_key.resolve_value(), url=self.api_base_url)
+
+        self.client = ModelInference(
+            model_id=self.model,
+            credentials=credentials,
+            project_id=self.project_id.resolve_value() if self.project_id else None,
+            verify=self.verify,
+            max_retries=self.max_retries,
+            delay_time=0.5,
+        )
+
+    def to_dict(self) -> dict[str, Any]:
+        """Serialize the component to a dictionary."""
+        return default_to_dict(
+            self,
+            model=self.model,
+            project_id=self.project_id.to_dict() if self.project_id else None,
+            api_base_url=self.api_base_url,
+            generation_kwargs=self.generation_kwargs,
+            api_key=self.api_key.to_dict(),
+            timeout=self.timeout,
+            max_retries=self.max_retries,
+            tools=self.tools,
+            verify=self.verify,
+        )
+
+    @classmethod
+    def from_dict(cls, data: dict[str, Any]) -> "WatsonxChatGenerator":
+        """Deserialize the component from a dictionary."""
+        deserialize_secrets_inplace(data["init_parameters"], keys=["api_key", "project_id"])
+        return default_from_dict(cls, data)
+
+    @component.output_types(replies=list[ChatMessage])
+    def run(
+        self,
+        messages: list[ChatMessage],
+        generation_kwargs: dict[str, Any] | None = None,
+        streaming_callback: StreamingCallbackT | None = None,
+    ) -> dict[str, list[ChatMessage]]:
+        """
+        Generate chat completions synchronously.
+
+        :param messages: List of ChatMessage objects
+        :param generation_kwargs: Additional generation parameters
+        :param streaming_callback: Callback function for streaming responses
+        :return: Dictionary with generated replies
+        """
+        if not messages:
+            return {"replies": []}
+
```

Comment:
In our Generators and ChatGenerators we always use the `select_streaming_callback` utility method so we can check if it's synchronous when used in `run` and async if used in `run_async`. 
It can be imported from here `from haystack.dataclasses import select_streaming_callback`
```python
        streaming_callback = select_streaming_callback(
            init_callback=self.streaming_callback, runtime_callback=streaming_callback, requires_async=False
        )
```